.github/workflows/image-push-master.yml

```diff
@@ -27,7 +30,98 @@ jobs:
-            ${{ env.IMAGE_NAME }}:latest
-            ${{ env.IMAGE_NAME }}:${{ github.sha }}
+            ${{ env.IMAGE_NAME }}:latest-amd64
           file: ./Dockerfile
+
+  build-and-push-arm64-sriov-cni:
+    name: Image Push ARM64
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push sriov-cni
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          platforms: linux/arm64
+          tags: |
+            ${{ env.IMAGE_NAME }}:latest-arm64
+          file: ./Dockerfile.arm64
+
+  build-and-push-ppc64le-sriov-cni:
+    name: Image Push ppc64le
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push sriov-cni
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          platforms: linux/ppc64le
+          tags: |
+            ${{ env.IMAGE_NAME }}:latest-ppc64le
+          file: ./Dockerfile.ppc64le
+
+  push-manifest:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        arch: [amd64,arm64,ppc64le]
+    needs: [build-and-push-amd64-sriov-cni,build-and-push-amr64-sriov-cni,build-and-push-ppc64le-sriov-cni]
+    steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Docker meta
+      id: docker_meta
+      uses: crazy-max/ghaction-docker-meta@v1
+      with:
+        images: ${{ env.IMAGE_NAME }}
+        tag-latest: false
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Create manifest for multi-arch images
+      run: |
+        docker pull ${{ env.IMAGE_NAME }}:latest-${{ matrix.arch }}
+        docker manifest create ${{ env.IMAGE_NAME }}:latest ${{ env.IMAGE_NAME }}:latest-${{ matrix.arch }}
+        docker manifest annotate ${{ env.IMAGE_NAME }}:latest ${{ env.IMAGE_NAME }}:latest-${{ matrix.arch }} --arch ${{ matrix.arch }}
+        docker manifest push ${{ env.IMAGE_NAME }}:latest
```

Comment:
after this bit is executed for amd64, arm64, ppc64le will we end up with a mainfest annotated with images for the 3 archs ?

just want to make sure, since in each iteration we create a new mainfest an push.

Reply:
Yes! I totally following to the multus

https://github.com/k8snetworkplumbingwg/multus-cni/blob/e692127d19623c8bdfc4d391224ea542658b584c/.github/workflows/image-push-release.yml#L235

Reply:
cool thx